hoodie-common/src/main/java/com/uber/hoodie/common/util/collection/DiskBasedMap.java

```diff
@@ -74,6 +73,22 @@ protected DiskBasedMap(String baseFilePath) throws IOException {
     this.filePosition = new AtomicLong(0L);
   }
 
+  /**
+   * RandomAcessFile is not thread-safe. This API opens a new file handle and returns.
+   * @return
+   */
+  private RandomAccessFile getNewRandomAccessFile()  {
```

Comment:
Instead of this, can we just create a synchronized version of the RandomAccessFile ? Wrap the class and implement 1 writer-N readers model, just like something discussed above for AbstractFileSystem..